core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -653,40 +650,26 @@ public void closeUfsBlock(long sessionId, long blockId)
   }
 
   @Override
-  public BlockReader createBlockReader(BlockReadRequest request) throws
-      BlockDoesNotExistException, IOException {
-    long sessionId = request.getSessionId();
-    long blockId = request.getId();
-    RetryPolicy retryPolicy = new TimeoutRetry(UFS_BLOCK_OPEN_TIMEOUT_MS, Constants.SECOND_MS);
-    while (retryPolicy.attempt()) {
-      try {
-        BlockReader reader = createLocalBlockReader(sessionId, blockId, request.getStart());
-        Metrics.WORKER_ACTIVE_CLIENTS.inc();
-        return reader;
-      } catch (BlockDoesNotExistException e) {
-        boolean checkUfs =
-            request.isPersisted() || (request.getOpenUfsBlockOptions() != null && request
-                .getOpenUfsBlockOptions().hasBlockInUfsTier() && request.getOpenUfsBlockOptions()
-                .getBlockInUfsTier());
-        if (!checkUfs) {
-          throw e;
-        }
-      }
-      // When the block does not exist in Alluxio but exists in UFS, try to open the UFS block.
-      try {
-        BlockReader reader = createUfsBlockReader(request.getSessionId(),
-            request.getId(), request.getStart(),
-            request.isPositionShort(), request.getOpenUfsBlockOptions());
-        Metrics.WORKER_ACTIVE_CLIENTS.inc();
-        return reader;
-      } catch (Exception e) {
```

Comment:
why removing this try-catch?

Reply:
This catches the IOException thrown from `createUfsBlockReader` and throw `UnavailableException` again. I just changed `createUfsBlockReader` to directly throw this exception instead. If a function is not handling an exception, it should not need to catch it.